sqlglot/dialects/athena.py

```diff
@@ -20,12 +20,37 @@ def _generate_as_hive(expression: exp.Expression) -> bool:
         else:
             return expression.kind != "VIEW"  # CREATE VIEW is never Hive but CREATE SCHEMA etc is
 
-    elif isinstance(expression, exp.Alter) or isinstance(expression, exp.Drop):
-        return True  # all ALTER and DROP statements are Hive
+    # https://docs.aws.amazon.com/athena/latest/ug/ddl-reference.html
+    elif isinstance(expression, (exp.Alter, exp.Drop, exp.Describe)):
+        if isinstance(expression, exp.Drop) and expression.kind == "VIEW":
+            # DROP VIEW is Trino (I guess because CREATE VIEW is)
+            return False
+
+        # Everything else is Hive
+        return True
 
     return False
 
 
+def _location_property_sql(self: Athena.Generator, e: exp.LocationProperty):
+    # If table_type='iceberg', the LocationProperty is called 'location'
+    # Otherwise, it's called 'external_location'
+    # ref: https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html
+
+    is_iceberg = False
+    if isinstance(e.parent, exp.Properties):
+        for rendered_prop in [
+            p.sql(dialect=self.dialect) for p in e.parent.expressions if isinstance(p, exp.Property)
```

Comment:
this is only used for iceberg?

could you do like p.name or something?

Reply:
Sure, i've refactored it to use `p.name`.

First, it needs to find the `exp.Property` object that represents the `table_type` property. Then it needs to extract the `value` member from that object and see if that expression equates to "iceberg".

The easy way out was to render the property and then check if both `table_type` and `iceberg` were in the resulting string